### DISCUSSION TITLE
Connected git repository not updating

### DISCUSSION THREAD

**Discussion by richleach** [vercel/community#1916] [Help] [answered]
### Summary

In my project settings, in the Git section, I have updated my project to point to a different git repo. It appears that the changes have been saved and recognized by Vercel but upon deployment my project is still pointing to the old repo, as stated in the Deployment Status logs. (I've double checked the branch name and have no Deploy Hooks). 

When doing a "Redeploy" I'm certain NOT to check the Redeploy with Existing Build Cache checkbox. 

Something is sticking, and I'm not sure how to "unstick". Any ideas?

Thanks in advance,

Rich

**Comment by amyegan** [comment]
Hi @richleach. Are you using GitLab?

**Comment by richleach** [comment]
Hi Amy

No I’m using VS Code on Mac (MacOS BigSur) and Github.

Rich

**Comment by richleach** [comment]
Amy 
In the process of testing through this issue I created a test repo in Github and connected to it from Vercel. 
I continue to see the message in the UI that I'm connected to the correct repo.
However whenever I view the logs after attempting to deploy it continues to point to the old repo.
I then completely deleted the test repo, pointed Vercel again to use a different repo in my Github account and even with those actions I get the following message when I attempt to deploy:
<img width="453" alt="image" src="https://user-images.githubusercontent.com/1168807/228619118-09b5943b-0d92-4e59-a649-a5008885e153.png">
Maybe I'm missing a step when I update my repo in the Settings section? 

Rich

**Reply by amyegan** [reply]
Let's rule out your hypothesis of missing steps before we dig into the possibility of trickier problems. 🙂 Can you tell me the exact steps you used to disconnect the old repo and connect a new one?

**Reply by richleach** [reply]
I log into my account on Vercel, go into my desired project. In Project Settings I click on Git on the left side. I'm presented with this screen: 
<img width="901" alt="image" src="https://user-images.githubusercontent.com/1168807/228655857-e71a36d7-2c30-4d35-9e3a-6dcc581993a6.png">
I then click on the Disconnect button, I'm presented with this screen (after I click Continue on the warning modal):
<img width="710" alt="image" src="https://user-images.githubusercontent.com/1168807/228656305-909d10fe-017b-4f68-b6a1-fdab6329021e.png">
I click the Github button, I'm presented with this screen:
<img width="683" alt="image" src="https://user-images.githubusercontent.com/1168807/228656546-a914114f-9e5e-42b3-88d3-5aa44b653e4d.png">
I click the Connect button for a different repo, I'm presented with this screen:
<img width="712" alt="image" src="https://user-images.githubusercontent.com/1168807/228656731-2baad80f-98ae-4eab-b1a1-c0073c8ba243.png">
I have confirmation from the UI that my repo is connected:
<img width="715" alt="image" src="https://user-images.githubusercontent.com/1168807/228656962-55cd7909-95a7-44ba-af5f-1c366d451bce.png">

I then navigate into my project's deployments and click Redeploy, I'm presented with this screen: 
<img width="687" alt="image" src="https://user-images.githubusercontent.com/1168807/228657579-ddb439c7-e7ad-4527-9619-a9ae44da8974.png">

When I click the Redeploy button on the modal/pop up I get this error: 
<img width="524" alt="image" src="https://user-images.githubusercontent.com/1168807/228657842-119fa579-b633-4480-9ae4-2fab3e84844a.png">

...and I get that error because I already deleted the github repo I was previously using because I pointed Vercel to use the newly assigned repo.

Am I doing that correctly?

Thank you,

Rich

**Reply by amyegan** [reply]
Thanks for walking me through that, RIch. Those connection steps are correct. I believe the trouble with redeploying is indeed because the repo is gone. But you should not have any trouble creating _new_ deployments from the currently connected repo

**Reply by richleach** [reply]
NO!
The very nature of this conversation is that I'm not able to update the currently connected repo.
After I disconnect the repo from my project and point the Vercel project to a new Github repo it still references the old (now disconnected) repo.
Maybe we can start from a different angle.... How do I change the Github repo my Vercel project is using to deploy from?
Please don't close this ticket, this issue is still open!
Thank you,
Rich

**Comment by amyegan** [comment, marked as answer]
Based on everything you have shared, it looks like your new repo is connected. You cannot redeploy things that came from a different repo at this point. You can only create a new deployment from the new repo. Once deployments exist from the new repo, you will be able to redeploy those.

### How to switch to a new GitHub repo to a Vercel Project
1. In the Git settings for the project, click the "Disconnect" button next to the currently connect repo and follow the prompts to disconnect.
2. When presented with options to connect a new repo, click the "GitHub" button.
3. Click the "Connect" button next to the repo you want to connect to this project.
4. Your Git settings will now reflect the current connected repo. New deployments will come from this repo.

**Comment by hellotazim** [comment]
is it possible to change the repo through vercel rest-api? I need to change the current repository for already deployed project. any way please?

**Reply by amyegan** [reply]
There is not currently a [Vercel API endpoint](https://vercel.com/docs/rest-api/endpoints) for that. But you can change the repo on the project dashboard

https://vercel.com/docs/deployments/git/vercel-for-github#changing-the-github-repository-of-a-project

**Reply by hellotazim** [reply]
How can i do it dynamically? is CLI method will help?
